sdk/src/main/scala/com/horizen/block/MainchainBackwardTransferCertificate.scala

```diff
@@ -0,0 +1,137 @@
+package com.horizen.block
+
+import com.horizen.utils.{BytesUtils, ListSerializer, VarInt}
+import scorex.core.serialization.{BytesSerializable, ScorexSerializer}
+import scorex.util.serialization.{Reader, Writer}
+
+class MainchainBTCertificateOutput
+  (val pubKeyHash: Array[Byte],
+   val amount: Long)
+  extends BytesSerializable
+{
+  override type M = MainchainBTCertificateOutput
+
+  override def serializer: ScorexSerializer[MainchainBTCertificateOutput] = MainchainBTCertificateOutputSerializer
+
+  var fee: Long = 0
+
+  def originalAmount: Long = {
+    amount + fee
+  }
+}
+
+object MainchainBTCertificateOutput {
+  def parse(outputBytes: Array[Byte], offset: Int): MainchainBTCertificateOutput = {
+
+    var currentOffset: Int = offset
+
+    val amount: Long = BytesUtils.getReversedLong(outputBytes, currentOffset)
+    currentOffset += 8
+
+    currentOffset += 4
+
+    val pubKeyHash: Array[Byte] = BytesUtils.reverseBytes(outputBytes.slice(currentOffset, currentOffset + 32))
+
+    new MainchainBTCertificateOutput(pubKeyHash, amount)
+  }
+}
+
+object MainchainBTCertificateOutputSerializer
+  extends ScorexSerializer[MainchainBTCertificateOutput]
+{
+  override def serialize(output: MainchainBTCertificateOutput, w: Writer): Unit = {
+    w.putLong(output.amount)
+    w.putLong(output.fee)
+    w.putInt(output.pubKeyHash.length)
+    w.putBytes(output.pubKeyHash)
+  }
+
+  override def parse(r: Reader): MainchainBTCertificateOutput = {
+    val amount = r.getLong()
+    val fee = r.getLong()
+    val keySize = r.getInt()
+    val pubKeyHash = r.getBytes(keySize)
+    val output = new MainchainBTCertificateOutput(pubKeyHash, amount)
+    output.fee = fee
+    output
+  }
+}
+
+class MainchainBackwardTransferCertificate
+  (val certificateBytes: Array[Byte],
+   val version: Int,
+   val sidechainId: Array[Byte],
+   val epochNumber: Int,
+   val endEpochBlockHash: Array[Byte],
+   val totalAmount: Long,
+   val outputs: Seq[MainchainBTCertificateOutput])
+  extends BytesSerializable
+{
+  override type M = MainchainBackwardTransferCertificate
+
+  override def serializer: ScorexSerializer[MainchainBackwardTransferCertificate] = MainchainBackwardTransferCertificateSerializer
+
+  def size: Int = certificateBytes.length
+
+}
+
+object MainchainBackwardTransferCertificate {
+  def parse(certificateBytes: Array[Byte], offset: Int) : MainchainBackwardTransferCertificate = {
```

Comment:
Add the following Unit tests asap:
1. Certificate with empty withdrawal requests
2. Certificate with non empty withdrawal requests
Verify all the fields inside the certificates.

Reply:
Still actual. Need to have a separate Unit tests for `MainchainBackwardTransferCertificate`